lib/haml_lint/linter/indentation.rb

```diff
@@ -9,15 +9,47 @@ class Linter::Indentation < Linter
       tab: /^\t*(?![ ])/,
     }.freeze
 
+    LEADING_SPACES_REGEX = /^( +)(?! )/
+
     def visit_root(root)
-      regex = INDENT_REGEX[config['character'].to_sym]
+      character = config['character'].to_sym
+      check_character(character, root)
+
+      width = config['width'].to_i
+      check_width(width, root) if character == :space && width > 0
```

Comment:
I like how you're breaking the character check apart from the width check, but I noticed that you're going to be traversing the tree of every file twice because of it.

If possible, I would like to see this in one pass.